modules/custom/openy_repeat/src/Controller/RepeatController.php

```diff
@@ -167,18 +239,31 @@ public function getLocationsInfo() {
    * Get detailed info about Class.
    */
   public function getClassesInfo() {
-    $nids = \Drupal::entityQuery('node')
-      ->condition('type','class')
-      ->execute();
-    $classes = Node::loadMultiple($nids);
-
-    $data = [];
-    foreach ($classes as $node) {
-      $data[$node->nid->value] = [
-        'nid' => $node->nid->value,
-        'title' => $node->title->value,
-        'description' => strip_tags(text_summary($node->field_class_description->value, $node->field_class_description->format, 600)),
-      ];
+    $data = $tags = [];
+    $cid = 'openy_repeat:classes_info';
+    if ($cache = $this->cache->get($cid)) {
+      $data = $cache->data;
+    }
+    else {
+      $nids = $this->entityQuery
+        ->get('node')
+        ->condition('type','class')
+        ->execute();
+      $nids_chunked = array_chunk($nids, 20, TRUE);
+      foreach ($nids_chunked as $chunk) {
+        $classes = $this->entityTypeManager->getStorage('node')->loadMultiple($chunk);
+        if (!empty($classes)) {
+          foreach ($classes as $node) {
+            $data[$node->nid->value] = [
+              'nid' => $node->nid->value,
+              'title' => $node->title->value,
+              'description' => html_entity_decode(strip_tags(text_summary($node->field_class_description->value, $node->field_class_description->format, 600))),
+            ];
+            $tags[] = 'node:' . $node->nid->value;
+          }
+        }
+      }
+      $this->cache->set($cid, $data, CacheBackendInterface::CACHE_PERMANENT, $tags);
```

Comment:
please add tag 'node_list' so whenever new class being added this cache also will be invalidated.